elastic/kibana
# OpenAI Preconfigured Connector Dashboard Link

## Summary

When I implemented the [dashboard to track OpenAI token usage](https://github.com/elastic/kibana/pull/159075), I did not consider preconfigured connectors. This PR adds a link from preconfigured edit connector flyout to token dashboard for OpenAI connectors.

<img width="1358" alt="Screenshot 2023-10-17 at 10 31 50 AM" src="https://github.com/elastic/kibana/assets/6935300/a765366c-c5be-48a0-ab2d-1fd2533dbf99">

## To test

1. Install an OpenAI to your `kibana.dev.yml`. You can use [these credentials](https://p.elstc.co/paste/1tUbSDcN#+5T5BbGKKlPZOHHDuO0apTYfIxPnzpXnwuWdhOWGJSb)
2. Go to Stack Management > Connectors
3. Select the preconfigured connector
4. Use the "Test" tab to send a message so there will be data in your dashboard (or don't - dashboard will work either way)
5. On the "Configuration" tab, click the link that says `View Usage Dashboard...`
6. Observe the dashboard.
